src/App.tsx

```diff
@@ -1,61 +1,136 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+
+function getUser(userId: number) {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser;
+}
+
+const initialTodos: Todo[]
+  = todosFromServer.map(
+    todo => ({
+      ...todo, user: getUser(todo.userId),
+    }),
+  );
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [todos, setTodos] = useState(initialTodos);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const addNewTodo = () => {
+    const idList = todos.map(
+      todo => todo.id,
+    );
+
+    const maxIdPlusOne = Math.max(...idList) + 1;
+
+    const newTodo: Todo = {
+      id: maxIdPlusOne,
+      title,
+      completed: false,
+      userId,
+      user: getUser(userId),
+    };
+
+    setTodos((prevTodos) => ([...prevTodos, newTodo]));
+
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserError(userId === 0);
+
+    if (title && userId > 0) {
```

Comment:
you should check the trimmed value to avoid such behavior:
![image](https://github.com/mate-academy/react_add-todo-form/assets/86923089/8961f5d4-ccee-4def-b58b-06858da405f8)